### DISCUSSION TITLE
Local Setup Failed

### DISCUSSION THREAD

**Discussion by jrTilak** [shadcn-ui/ui#2328] [General]
Hey, I am trying to run the project locally and I am getting some errors.

```bash
PS C:\Users\sagar\OneDrive\Desktop\OSC\shadcn-ui-forked> pnpm --filter=www dev

> www@0.0.1 dev C:\Users\sagar\OneDrive\Desktop\OSC\shadcn-ui-forked\apps\www
> next dev -p 3001

node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error: Cannot find package 'C:\Users\sagar\OneDrive\Desktop\OSC\shadcn-ui-forked\node_modules\.pnpm\@effect-ts+otel-exporter-trace-otlp-grpc@0.15.1_@effect-ts+core@0.60.5_@opentelemetry+api@1.4_aul557yiuu22b5xwpe7cspv3aa\node_modules\@opentelemetry\exporter-trace-otlp-grpc\package.json' imported from C:\Users\sagar\OneDrive\Desktop\OSC\shadcn-ui-forked\node_modules\.pnpm\@effect-ts+otel-exporter-trace-otlp-grpc@0.15.1_@effect-ts+core@0.60.5_@opentelemetry+api@1.4_aul557yiuu22b5xwpe7cspv3aa\node_modules\@effect-ts\otel-exporter-trace-otlp-grpc\_mjs\index.mjs
Did you mean to import @opentelemetry+exporter-trace-otlp-grpc@0.39.1_@opentelemetry+api@1.4.1/node_modules/@opentelemetry/exporter-trace-otlp-grpc/build/src/index.js?
    at legacyMainResolve (node:internal/modules/esm/resolve:205:26)
    at packageResolve (node:internal/modules/esm/resolve:831:14)
    at moduleResolve (node:internal/modules/esm/resolve:901:20)
    at defaultResolve (node:internal/modules/esm/resolve:1121:11)
    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:396:12)
    at ModuleLoader.resolve (node:internal/modules/esm/loader:365:25)
    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:240:38)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:85:39)
    at link (node:internal/modules/esm/module_job:84:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v20.10.0
C:\Users\sagar\OneDrive\Desktop\OSC\shadcn-ui-forked\apps\www:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  www@0.0.1 dev: `next dev -p 3001`
Exit status 1
```

What I did?

=> I tried to run the project as mentioned [here](https://github.com/shadcn-ui/ui/blob/main/CONTRIBUTING.md).

But, when i run `pnpm --filter=www dev`, I am getting this error.

![image](https://github.com/shadcn-ui/ui/assets/106688422/32e83351-e434-4da8-bd14-ec9a42f104d5)

Thanks!!

**Comment by Febin-Joseph** [comment]
Hi @jrTilak, I think problem you might be occuring due to the version of Node js. in your error it is showing your Node js version as Node.js v20.10.0 the latest version.
But if using Node js version 18.17.0  i think the problem might be fixed .
for using or changing the current version of `Node.js v20.10.0`    to   `Node.js v18.17.0`  use can use nvm `Node Version Manager` for switching the current version to the desired Version of Node js .

STEPS :
1) So for using nvm you can download it from here  [nvm-setup.exe](https://github.com/coreybutler/nvm-windows/releases/download/1.1.12/nvm-setup.exe)
2) And after the installation use this command in Terminal or cmd `nvm install 18.17.0`
3) After that use `nvm use 18.17.0`
4) After that check your node js version by `node -v` check is changed to v18.17.0


And let me know you problem is now has been fixed or not...

**Reply by jrTilak** [reply]
A 1000x Thanks 🫡, It worked.

**Comment by jrTilak** [comment]
Thanks @Febin-Joseph  for the solution. By changing the node version, it worked.

By the way, can you explain the solution? I am little curious, how it worked just by changing the node version.

**Reply by Febin-Joseph** [reply]
Glad it worked,
ya sure , i will explain that, the problem is not your's the problem is occuring because some Node.js version is not compatible with the version of Next.js so we needed a minimum of `v18.17.0`  and also i think dont wanted to be use the latest Node js version like you where using `Node.js v20.10.0` we needed to use a version between them.

**Comment by i-m-abbhay** [comment]
Thanks Guys @Febin-Joseph @jrTilak . :) The same problem was occurring on my end. I am glad that I ran into this thread.
